Домашнее задание к лекции «Случайные события. Случайные величины»

Смоделировать игру против лотерейного автомата типа “777”. Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш (см. справа)

Выгодна ли игра игроку?

Сколько в среднем приобретает
или теряет игрок за одну игру?

777: 200 руб.
999: 100 руб.
555: 50 руб.
333: 15 руб.
111: 10 руб.
*77: 5 руб.
**7: 3 руб.
*00: 2 руб.
**0: 1 руб.

– любая цифра

In [14]:
import numpy as np

In [33]:
trials = 1000000

In [34]:
profit = 0

In [35]:
successes = 0

In [36]:
for x in range(trials):
    x = np.random.randint(0, 1000)    
    if x == 777:
        successes += 1
        profit += 200
    elif x == 999:
        successes += 1
        profit += 100
    elif x == 555:
        successes += 1
        profit += 50
    elif x == 333:
        successes += 1
        profit += 15
    elif x == 111:
        successes += 1
        profit += 10
    elif x % 100 == 77:
        successes += 1
        profit += 5
    elif x % 10 == 7:
        successes += 1
        profit += 3
    elif x % 100 == 0:
        successes += 1
        profit += 2
    elif x % 10 == 0:
        successes += 1
        profit += 1
print(f'Количество выигрышных попыток: {successes}')
print(f'Выигрыш составил: {profit} руб.')      
 
       

        
    

Количество выигрышных попыток: 203643
Выигрыш составил: 798317 руб.


In [38]:
#Средний выигрыш игрока за игру
profit/trials

0.798317

In [39]:
#Баланс игрока за одну попытку:
1 - profit/trials

0.20168299999999995

Игра невыгодна для игрока. Было потрачено всего 1 миллион рублей, выигрыш составил 798317 рублей. В среднем за одну игру игрок теряет около 20 копеек.